auth/src/main/java/com/vimeo/networking2/Authenticator.kt

```diff
@@ -31,31 +29,23 @@ class Authenticator(private val serverConfig: ServerConfig) {
      */
     private var retrofitServicesCache: RetrofitServicesCache
 
+    private var apiErrorResponseErrorConverter: ApiResponseErrorConverter
+
     init {
         retrofit = RetrofitSetupComponent.retrofit(serverConfig)
         retrofitServicesCache = RetrofitSetupComponent.retrofitCache(retrofit)
+        apiErrorResponseErrorConverter = RetrofitSetupComponent.apiResponseErrorConverter(retrofit)
     }
 
-    /**
-     * This is a temporary request that is added in to show how the access token will be set.
-     * This will be removed in the next PR which is to design and implement all the auth requests.
-     */
-    fun authenticateWithClientCredentials() {
-        val apiService = retrofitServicesCache.getService(AuthService::class.java)
-
-        val call = apiService.authorizeWithClientCredentialsGrant(
+    fun clientCredentials(): ClientCredentialsAuthenticator {
+        return ClientCredentialsInteractor(
+            apiErrorResponseErrorConverter,
+            retrofitServicesCache.getService(AuthService::class.java),
             RetrofitSetupComponent.authHeaders(serverConfig.clientId, serverConfig.clientSecret),
-            GrantType.CLIENT_CREDENTIALS.value,
-            ScopeType.PUBLIC.value)
-
-        call.enqueue(object : Callback<VimeoAccount> {
-
-            override fun onResponse(call: Call<VimeoAccount>, response: Response<VimeoAccount>) {
-                response.body()?.accessToken?.let { setAccessToken(it) }
-            }
-
-            override fun onFailure(call: Call<VimeoAccount>, t: Throwable) {}
-        })
+            serverConfig.scopes.joinToString { it.value }
+        ) {
+            setAccessToken(it)
```

Comment:
Optional (and maybe a personal pref) but I find prefer using last-arg lambads like this when something is being executed and we're expecting a response.

eg:
```kotlin
makeNetworkRequest(scope) { response ->
 ...
}
```

But for constructors it's weird because in a case like this where the lambda executed after a call to `authenticate`, and only on success, this makes it so that it's not obvious what the lambda is doing or when it's going to be called, so a dev needs to dig into the implementation to find out.

simple eg:
```kotlin
NetworkRequestor(scope, url, blah, blah) {
  8080
}
```
here you would need to go into the impl to find that this is some callback to get a port number, why it's needed, and when it's used.

If you agree with this, I'd suggest adding this lambda to the params instead of hanging it off the end.

i.e.:
```kotlin
return ClientCredentialsInteractor(
            apiResponseErrorConverter = apiErrorResponseErrorConverter,
            authService = retrofitServicesCache.getService(AuthService::class.java),
            authHeaders = RetrofitSetupComponent.authHeaders(serverConfig.clientId, serverConfig.clientSecret),
            scopes = serverConfig.scopes.joinToString { it.value },
            authTokenCallback = { setAccessToken(it) }
        )
```

... to me this makes things more clear when & why it will be used.